src/lavinmq/message_store.cr

```diff
@@ -463,9 +463,11 @@ module LavinMQ
 
     private def read_metadata_file(seg, mfile)
       metafile = meta_file_name(mfile)
-      count = File.open(metafile) do |f|
-        @replicator.try &.register_file(f)
-        f.read_bytes(UInt32)
+      count = File.open(metafile) do |file|
+        msg_count = file.read_bytes(UInt32)
+        @replicator.try &.register_file(file)
+        read_extra_metadata_fields(file, seg)
```

Comment:
It feels a little off to adapt the parent class to work for those inheriting it.. On the other hand I do like that you got rid of the duplicate code! maybe a comment in the empty `read_extra_metadata_fields` function to explain so we avoid getting confused about this in the future?

Reply:
>It feels a little off to adapt the parent class to work for those inheriting it.. 

I agree, but it seems like its the best option given the current structure. What we could do is to make an abstract class or similar that does the basic shared stuff and then have the message_store and stream_message_store inherit from that and then have those classes decide what they want to do in the `read_extra_metadata_fields` method. But I don't think that is something we would want to do?

Can surely add a comment if we go with this approach. If we take the approach I mention here, we would probably not need the comment, but some extra work hehe :D

Reply:
I think it's fine. It's some kind of template method pattern.

Reply:
Yeah sounds good, go with this! :)